### PR TITLE
Add a matcher for Matrix4 that includes epsilon

### PR DESCRIPTION
**Pull request by jonahwilliams**
To make patterns like https://github.com/flutter/plugins/pull/6088#pullrequestreview-1033292266 easier, and encourage folks to compare doubles with an epsilon.

### CODE REVIEW

File: packages/flutter_test/lib/src/matchers.dart [hunk 1]
```diff
@@ -266,6 +266,18 @@ Matcher rectMoreOrLessEquals(Rect value, { double epsilon = precisionErrorTolera
...
 
+/// Asserts that two [Matrix4]s are equal, within some tolerated error.
+///
+/// {@macro flutter.flutter_test.moreOrLessEquals}
+///
+/// See also:
+///
+///  * [moreOrLessEquals], which is for [double]s.
+///  * [offsetMoreOrLessEquals], which is for [Offset]s.
+Matcher matrixMorOrLessEqual(Matrix4 value, { double epsilon = precisionErrorTolerance }) {
```
Comment (fzyzcjy) [inline packages/flutter_test/lib/src/matchers.dart:277]:
```suggestion
Matcher matrixMoreOrLessEqual(Matrix4 value, { double epsilon = precisionErrorTolerance }) {
```

Comment (fzyzcjy) [reply packages/flutter_test/lib/src/matchers.dart:277]: uber-nit

Comment (jonahwilliams) [reply packages/flutter_test/lib/src/matchers.dart:277]: oops

Comment (fzyzcjy) [reply packages/flutter_test/lib/src/matchers.dart]: You are welcome!

**Review by cyanglaz — APPROVED** [review]
LGTM

**Comment by cyanglaz** [comment]
https://github.com/flutter/plugins/pull/6088 can be improved after this lands in stable.

**Pull request merged**